### PR TITLE
Fix for data race in heartBeatTimeChecker tests

### LINKED ISSUE #325: data race during testing on NodeAgent tests
A repro here: https://github.com/PlayFab/thundernetes/runs/7438045505?check_suite_focus=true

Occasionally, there will be a data race when running the NodeAgent tests. We suspect this happens because

- we manually set the nowFunc method to mock time https://github.com/PlayFab/thundernetes/blob/main/cmd/nodeagent/nodeagentmanager_test.go#L639 and, at the same time,
- we read it on the (periodically running) HeartbeatTimeChecker on https://github.com/PlayFab/thundernetes/blob/main/cmd/nodeagent/nodeagentmanager.go#L118

This failure cannot repro locally, since (we think) the tests run before the HeartbeatTimeChecker has a chance to execute.

### PR DESCRIPTION
**Pull request by javier-op**
Fixes #325 
Added parameter to turn off the heartBeatTimeCheckerLoop, so you can run it manually without creating a data race.

### CODE REVIEW

File: cmd/nodeagent/nodeagentmanager_test.go [hunk 1]
```diff
@@ -809,7 +809,7 @@ var _ = Describe("nodeagent tests", func() {
 		// change time to be 6 seconds later again
```
Comment (dgkanatsios) [inline cmd/nodeagent/nodeagentmanager_test.go:809]: should we change this comment as well?

File: cmd/nodeagent/nodeagentmanager_test.go [hunk 2]
```diff
@@ -41,7 +41,7 @@ var _ = Describe("nodeagent tests", func() {
 	It("heartbeat with empty body should return error", func() {
 		req := httptest.NewRequest(http.MethodPost, "/v1/sessionHosts/sessionHostID", nil)
 		w := httptest.NewRecorder()
-		n := NewNodeAgentManager(newDynamicInterface(), testNodeName, false, false, time.Now)
+		n := NewNodeAgentManager(newDynamicInterface(), testNodeName, false, false, time.Now, true)
```
Comment (dgkanatsios) [inline cmd/nodeagent/nodeagentmanager_test.go:44]: do some tests need the checker to run while some others don't?

File: cmd/nodeagent/nodeagentmanager.go [hunk 3]
```diff
@@ -72,7 +72,11 @@ func NewNodeAgentManager(dynamicClient dynamic.Interface, nodeName string, logEv
 		nowFunc:                   now,
 	}
 	n.runWatch()
-	n.runHeartbeatTimeCheckerLoop()
+	n.firstHeartbeatTimeout = ParseInt64FromEnv("FIRST_HEARTBEAT_TIMEOUT", 60000)
```
Comment (dgkanatsios) [inline cmd/nodeagent/nodeagentmanager.go:75]: curious, why are these moved out of the method?

File: cmd/nodeagent/main.go [hunk 4]
```diff
@@ -28,7 +28,7 @@ func main() {
 		log.Fatal(err)
 	}
 
-	n := NewNodeAgentManager(dynamicClient, nodeName, logEveryHeartbeat, ignoreHealthFromHeartbeat, time.Now)
+	n := NewNodeAgentManager(dynamicClient, nodeName, logEveryHeartbeat, ignoreHealthFromHeartbeat, time.Now, true)
```
Comment (dgkanatsios) [inline cmd/nodeagent/main.go:31]: idea, maybe for a different PR: add an option (command line flag?) so user can turn it on or off (default should be on though)

File: cmd/nodeagent/nodeagentmanager_test.go (hunk 1 above)
Comment (javier-op) [reply cmd/nodeagent/nodeagentmanager_test.go:809]: No it's okay, the previous time was 10:00:06, so i wasn't actually changing the time again.

File: cmd/nodeagent/nodeagentmanager_test.go (hunk 2 above)
Comment (javier-op) [reply cmd/nodeagent/nodeagentmanager_test.go:44]: There are 5 heartbeat tests, and in 3 of these i run the checker manually, so i turned the loop off in these cases.

**Review by dgkanatsios — APPROVED** [review]

**Pull request merged**